deepchem/utils/molecule_graph.py

```diff
@@ -0,0 +1,187 @@
+  num_node_features : int
+    The number of features per node in the graph
+  num_edges : int
+    The number of edges in the graph
+  num_edges_features : int, , optional (default None)
+    The number of features per edge in the graph
+  """
+
+  def __init__(
+      self,
+      node_features: np.ndarray,
+      edge_index: np.ndarray,
+      targets: np.ndarray,
+      edge_features: Optional[np.ndarray] = None,
+      graph_features: Optional[np.ndarray] = None,
+  ):
+    """
+    Parameters
+    ----------
+    node_features : np.ndarray
+      Node feature matrix with shape [num_nodes, num_node_features]
+    edge_index : np.ndarray, dtype int
+      Graph connectivity in COO format with shape [2, num_edges]
+    targets : np.ndarray
+      Graph or node targets with arbitrary shape
+    edge_features : np.ndarray, optional (default None)
+      Edge feature matrix with shape [num_edges, num_edge_features]
+    graph_features : np.ndarray, optional (default None)
+      Graph feature vector with shape [num_graph_features,]
+    """
+    # validate params
+    if isinstance(node_features, np.ndarray) is False:
+      raise ValueError('node_features must be np.ndarray.')
+    if isinstance(edge_index, np.ndarray) is False:
+      raise ValueError('edge_index must be np.ndarray.')
+    elif edge_index.dtype != np.int:
+      raise ValueError('edge_index.dtype must be np.int')
+    elif edge_index.shape[0] != 2:
+      raise ValueError('The shape of edge_index is [2, num_edges].')
+    if isinstance(targets, np.ndarray) is False:
+      raise ValueError('y must be np.ndarray.')
+    if edge_features is not None:
+      if isinstance(edge_features, np.ndarray) is False:
+        raise ValueError('edge_features must be np.ndarray or None.')
+      elif edge_index.shape[1] != edge_features.shape[0]:
+        raise ValueError('The first dimension of edge_features must be the \
+                    same as the second dimension of edge_index.')
+    if graph_features is not None and isinstance(graph_features,
+                                                 np.ndarray) is False:
+      raise ValueError('graph_features must be np.ndarray or None.')
+
+    self.node_features = node_features
+    self.edge_index = edge_index
+    self.edge_features = edge_features
+    self.graph_features = graph_features
+    self.targets = targets
+    self.num_nodes, self.num_node_features = self.node_features.shape
+    self.num_edges = edge_index.shape[1]
+    if self.node_features is not None:
+      self.num_edge_features = self.edge_features.shape[1]
+
+  def to_pyg_data(self):
+    """Convert to PyTorch Geometric Data instance
+
+    Returns
+    -------
+    torch_geometric.data.Data
+      Molecule graph data for PyTorch Geometric
+    """
+    try:
+      import torch
+      from torch_geometric.data import Data
+    except ModuleNotFoundError:
+      raise ValueError(
+          "This class requires Pytorch and PyTorch Geometric to be installed.")
```

Comment:
Can you swap this out to say "This method..." since the rest of the class doesn't require PyTorch to be implemented?

Can you also add PyTorch Geometric onto the lift of soft-dependencies and update the dependency table in the docs?

Reply:
I fixed!